src/languages/yaml.js

```diff
@@ -39,7 +39,7 @@ export default function(hljs) {
     variants: [
       {begin: /'/, end: /'/},
       {begin: /"/, end: /"/},
-      {begin: /\S+/}
+      {begin: /[^\s,{}[\]]+/}
```

Comment:
I'm not sure about this.  What is the correct behavior for:

```
foo: I am a string { boo: hoo } 
```

Also I'm not sure this is the desired fix in any case.  What you'd really want to do is match on `[` or `{` and then go into an alternative mode (to allow matching embedded strings, numbers, etc)... not overload string in this fashion.

Reply:
This change simply makes it so strings do *not* match braces or brackets or commas. I don't think your example is valid yaml since it mixes two different types of nodes (they must be one of scalar, sequence, or mapping).

Reply:
I don't think that what is valid inside of braces or brackets is any different from outside, except  that the brackets and braces must of course be matching, and commas are only valid inside.

Reply:
It seems these are certainly characters that are valid inside strings:

```rb
> YAML.load("---\n:test: boo[]ger{mode:[32]}").to_h
=> {:test=>"boo[]ger{mode:[32]}"}
```

Again negative matching isn't the way to go about this.  You need to detect `[` or `{` and then switch modes.

Reply:
Again, looking at the JSON stuff might be a big help.

Reply:
> I don't think that what is valid inside of braces or brackets is any different from outside, except that the brackets and braces must of course be matching, and commas are only valid inside.

Right but if you don't want them to be counted as strings then you need to explain to the ruleset what they are. :-)  Since the general case is "anything following a key is a string"... these are exception cases and must be accounted as such.

And surely what is valid inside is a little different... at least regarding line breaks, yes?